hdkeychain/extendedkey.go

```diff
@@ -296,7 +296,7 @@ func (k *ExtendedKey) Child(i uint32) (*ExtendedKey, error) {
 		ilNum.Mod(ilNum, btcec.S256().N)
 		childKey = ilNum.Bytes()
 		// Correction a key-length with leading zero
-		if len(childKey) < 32 {
+		if len(childKey) < 32 && len(fixLeadingZeroBug) == 1 && fixLeadingZeroBug[0] == true {
```

Comment:
why not `len(...) > 0`?

With this way, `extKey, err = extKey.Child(childNum, true, true, true)` will not work even though the second arg is true.

Reply:
> why not `len(...) > 0`?
> 
> With this way, `extKey, err = extKey.Child(childNum, true, true, true)` will not work even though the second arg is true.

I think this is not an explicit use and suggest not supporting it

Reply:
Then at the top of the method check that len(fixLeadingZeroBug) is 0 or 1, and if anything else, return an error.

If I pass true to the second arg I would expect it to work.

Reply:
but tbh go doesn't have optional parameters or overloading for a reason... so this isn't very go-like...

(I am not a mainly go programmer though)